lightly/models/modules/momentum.py

```diff
@@ -0,0 +1,83 @@
+""" Momentum Encoder """
+
+# Copyright (c) 2020. Lightly AG and its affiliates.
+# All Rights Reserved
+
+import copy
+
+import torch
+import torch.nn as nn
+
+@torch.no_grad()
+def batch_shuffle(batch: torch.Tensor):
+    """Returns the shuffled batch and the indices to undo.
+
+    Examples:
+        >>> # forward pass through the momentum model with batch shuffling
+        >>> x1_shuffled, shuffle = batch_shuffle(x1)
+        >>> f1 = moco_momentum(x1)
+        >>> out0 = projection_head_momentum(f0)
+        >>> out1 = batch_unshuffle(out1, shuffle)
+    """
+    batch_size = batch.shape[0]
+    shuffle = torch.randperm(batch_size, device=batch.device)
+    return batch[shuffle], shuffle
+
+@torch.no_grad()
+def batch_unshuffle(batch: torch.Tensor, shuffle: torch.Tensor):
+    """Returns the unshuffled batch.
+
+    Examples:
+        >>> # forward pass through the momentum model with batch shuffling
+        >>> x1_shuffled, shuffle = batch_shuffle(x1)
+        >>> f1 = moco_momentum(x1)
+        >>> out0 = projection_head_momentum(f0)
+        >>> out1 = batch_unshuffle(out1, shuffle)
+    """
+    unshuffle = torch.argsort(shuffle)
+    return batch[unshuffle]
+
+def deactivate_requires_grad(params):
+    """Deactivates the requires_grad flag for all parameters.
+    
+    This has the same effect as executing the model within a `torch.no_grad()`
```

Comment:
I would mention that it's like permanently executing the model within a `torch.no_grad()` context. 

Small typo: "Use this method to disable**d** gradient computation..."